pages/api/v1/trades/transactTrade/index.ts

```diff
@@ -0,0 +1,167 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { connect } from '../../../database/database'
+import _ from 'lodash'
+import { ObjectId } from 'mongodb'
+import { getAccessTokenFromHeader } from '../../common'
+
+let checkForCardsInCollection = (cardCollection, tradeCards) => {
+  // track indices of cards in trade with user's collection
+  let cardHash = {}
+  let cardArrayFromCollection = []
+
+  tradeCards.forEach((card) => {
+    // if card has already been found at least once
+    if (cardHash[card]) {
+      // update new "last found" position for card
+      cardArrayFromCollection.push(
+        cardCollection.find((element) => element._id === card)
+      )
+      cardHash[card] = cardCollection.indexOf(card, cardHash[card] + 1)
+    }
+
+    if (!cardHash[card]) {
+      cardHash[card] = cardCollection.indexOf(card)
+      cardArrayFromCollection.push(
+        cardCollection.find((element) => element._id === card)
+      )
+    }
+  })
+
+  // if an index is -1, the user does not have all of cards in the trade.
+  if (!Object.values(cardHash).every((value) => value !== -1)) {
+    return false
+  }
+
+  return cardArrayFromCollection
+}
+
+const index = async (request: NextApiRequest, response: NextApiResponse) => {
```

Comment:
This request in particular we'll want to make certain is working correctly. The `.splice` calls I'm making are modifying a user's card array and I don't want people to lose their cards.